statsd/statsd.go

```diff
@@ -120,36 +141,103 @@ func (c *Client) watch() {
 }
 
 func (c *Client) append(cmd string) error {
-	c.Lock()
 	c.commands = append(c.commands, cmd)
 	// if we should flush, lets do it
 	if len(c.commands) == c.bufferLength {
 		if err := c.flush(); err != nil {
-			c.Unlock()
 			return err
 		}
 	}
-	c.Unlock()
 	return nil
 }
 
+func (c *Client) joinMaxSize(cmds []string, sep string, maxSize int) ([][]byte, []int) {
+	c.buffer.Reset() //clear buffer
+
+	var frames [][]byte
+	var ncmds []int
+	sepBytes := []byte(sep)
+
+	elem := 0
+	for _, cmd := range cmds {
+		needed := len(cmd)
+
+		//if a single command is too big, flush it and hope the UDP fragmented packet
+		//doesn't get dropped.
+		if needed > maxSize {
```

Comment:
Sure, I see you pulled my changes to you pull request, do you want me to close this one?
You're right about commands ordering, maybe we can flush everything we had accumulated before flushing a big command, this would be less efficient, but would preserve order